### PR TITLE
Use gauge for last_saved_block_id S3 metric

### PR DESCRIPTION
**Pull request by tdimitrov**
Gauge is more suitable type for the metric as it represents a numeric
value.

### CODE REVIEW

File: storage/include/s3/metrics.hpp [hunk 1]
```diff
@@ -35,10 +36,23 @@ struct Metrics {
...
+      return;
+    } catch(std::exception& e) {
+      LOG_ERROR(logger_, "Unexpected error occured while converting lastSavedBlockId (" << elems[elems.size() - 2] << ") to numeric value.");
+      return;
+    }
+    lastSavedBlockId_.Get().Set(lastSavedBlockVal);
   }
 
-  std::string getLastSavedBlockId() { return lastSavedBlockId_.Get().Get(); }
+  uint64_t getLastSavedBlockId() { return lastSavedBlockId_.Get().Get(); }
```
Comment (f-squirrel) [inline storage/include/s3/metrics.hpp:55]: Nit: `BlockId`?

File: storage/include/s3/metrics.hpp [hunk 2]
```diff
@@ -35,10 +36,23 @@ struct Metrics {
     // the format is: "PREFIX/BLOCK_ID/KEY", where PREFIX is optional
     if (elems.size() < 2) return;
 
-    lastSavedBlockId_.Get().Set(elems[elems.size() - 2]);
+    uint64_t lastSavedBlockVal = 0;
```
Comment (f-squirrel) [inline storage/include/s3/metrics.hpp:39]: Nit: As far as I see, the file uses snake_case, so probably`last_saved_block_val`?
P.S. It is really frustrating that we have both...

Comment (f-squirrel) [reply storage/include/s3/metrics.hpp:39]: Actually, it is mixed... so ignore, please.

Comment (tdimitrov) [reply storage/include/s3/metrics.hpp:40]: Good point. I fixed the naming in the whole file. It's a new source so it should be styled correctly.

**Pull request merged**